server/service/src/sync/translations/name_category_5.rs

```diff
@@ -0,0 +1,251 @@
+                ActivityLogFilter::new().record_id(EqualFilter::equal_to(&changelog.record_id)),
+            )?
+            .into_iter()
+            .max_by_key(|log| log.activity_log_row.datetime);
+
+        let previous_values = match activity_log {
+            Some(log) => match log.activity_log_row.changed_from {
+                Some(ref s) => s.split(',').map(|v| v.trim().to_string()).collect(),
+                None => Vec::new(),
+            },
+            None => Vec::new(),
+        };
+
+        let current_values = match &property.allowed_values {
+            Some(values) => values
+                .split(',')
+                .map(|s| s.trim().to_string())
+                .collect::<Vec<_>>(),
+            None => {
+                return Ok(PushTranslateResult::Ignored(
+                    "No allowed values to push".to_string(),
+                ))
+            }
+        };
+
+        let removed_values: Vec<String> = previous_values
+            .into_iter()
+            .filter(|v| !current_values.contains(v))
+            .collect();
+
+        let mut records = Vec::new();
+
+        // Handle removed values (deletes)
+        for description in removed_values {
+            let id = format!("{}:{}", changelog.record_id, description);
+            let legacy_row = LegacyNameCategory5Row {
+                ID: id.clone(),
+                description: description.to_string(),
+                r#type: 'c'.to_string(),
+            };
+
+            records.push(PushSyncRecord {
+                cursor: changelog.cursor,
+                record: CommonSyncRecord {
+                    table_name: self.table_name().to_string(),
+                    record_id: id,
+                    action: SyncAction::Delete,
+                    record_data: serde_json::to_value(legacy_row)?,
+                },
+            });
+        }
+
+        // Handle current values (updates)
+        for description in current_values.iter() {
+            let id = format!("{}:{}", changelog.record_id, description);
+            let legacy_row = LegacyNameCategory5Row {
+                ID: id.clone(),
+                description: description.to_string(),
+                r#type: 'c'.to_string(),
+            };
+
+            records.push(PushSyncRecord {
+                cursor: changelog.cursor,
+                record: CommonSyncRecord {
+                    table_name: self.table_name().to_string(),
+                    record_id: id,
+                    action: SyncAction::Update,
+                    record_data: serde_json::to_value(legacy_row)?,
+                },
+            });
+        }
+
+        Ok(PushTranslateResult::PushRecord(records))
+    }
+
+    fn try_translate_from_delete_sync_record(
+        &self,
+        connection: &StorageConnection,
+        sync_buffer: &SyncBufferRow,
+    ) -> Result<PullTranslateResult, anyhow::Error> {
+        let mut parts = sync_buffer.record_id.splitn(2, ':');
```

Comment:
TODO: Create issue in OG to update the record_id returned in syntax: `id:supply_level`